### PR TITLE
Add MATLAB language 

### LINKED ISSUE #1818: [Feature Request]:  Adding new Language under docs
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

I would like to suggest adding the MATLAB programming language section to our website. 
1. Introduction 
2. Environment setup
3. variable
4. Commands
5. Data Type
6. Operators
7. Vector
8. Matrix
9. Array
10. String
 
Advance MATLAB 
1. Algebra
2. Plotting
3. Graphics

Uses /Application of MATLAB 
### Use Case

.

### Benefits

MATLAB is widely used in engineering, science, and mathematical computations, and having a dedicated section for it would greatly benefit our users.

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by Khushi-Pushkar**
## Related Issue

[Cite any related issue(s) this pull request addresses. If none, simply state "None”]

## Description

[Please include a brief description of the changes or features added]

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [x] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [ ] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [x] I have followed security best practices in my code changes.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [x] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [x] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [x] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [x] [Markdown Guide](https://www.markdownguide.org/)
- [x] [MDX Documentation](https://mdxjs.com/docs/)
- [x] [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- [x] [Tabs: Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [x] [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [x] [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex) 

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!
ADDED MATLAB language 
Kindly comment if any changes needed
Fix #1818

### CODE REVIEW

**Comment by Khushi-Pushkar** [comment]
@Ajay-Dhangar Please Review it soon

File: docs/MATLAB/Advance MATLAB/Poly.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
+---
+id: MATLAB Polynomials
```
Comment (Ajay-Dhangar) [inline docs/MATLAB/Advance MATLAB/Poly.md:2]: replace `id: MATLAB Polynomials` to `id: matlab-polynomials`

File: docs/MATLAB/Advance MATLAB/Poly.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+title: Polynomials
+sidebar_label: MATLAB Polynomials
+sidebar_position: 2
+tags: [MATLAB, Polynomials]
+description: In this tutorial, you will learn about Polynomials in MATLAB.
+---
+# MATLAB Polynomials
```
Comment (Ajay-Dhangar) [inline docs/MATLAB/Advance MATLAB/Poly.md:9]: remove `# MATLAB Polynomials`

File: docs/MATLAB/Advance MATLAB/alg.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
+---
+id: MATLAB Algebra
```
Comment (Ajay-Dhangar) [inline docs/MATLAB/Advance MATLAB/alg.md:2]: update `id` for good practice

File: docs/MATLAB/Advance MATLAB/alg.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+title: Algebra
+sidebar_label: MATLAB Algebra
+sidebar_position: 1
+tags: [MATLAB, Algebra]
+description: In this tutorial, you will learn about Algebra in MATLAB.
+---
+# MATLAB Algebra
```
Comment (Ajay-Dhangar) [inline docs/MATLAB/Advance MATLAB/alg.md:9]: remove this line

File: docs/MATLAB/Command.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
+---
+id: MATLAB Command
```
Comment (Ajay-Dhangar) [inline docs/MATLAB/Command.md:2]: update `id` for good practice

File: docs/MATLAB/Command.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+title: Command
+sidebar_label: MATLAB Command
+sidebar_position: 
+tags: [MATLAB, Command]
+description: In this tutorial, you will learn about Command in MATLAB.COMMAND in MATLAB is a keyword used in the creation and manipulation of command line interfaces (CLI) within MATLAB. 
+---
+# MATLAB Command
```
Comment (Ajay-Dhangar) [inline docs/MATLAB/Command.md:9]: remove this line

File: docs/MATLAB/Introduction.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
+---
+id: MATLAB
```
Comment (Ajay-Dhangar) [inline docs/MATLAB/Introduction.md:2]: update `id` for good practice

File: docs/MATLAB/Introduction.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+title: Introduction to MATLAB
+sidebar_label: An Introduction to MATLAB
+sidebar_position: 1
+tags: [MATLAB, Feature, Uses]
+description: "Learn Basics of MATLAB.MATLAB (Matrix Laboratory) is a high-performance language for technical computing. It integrates computation, visualization, and programming in an easy-to-use environment where problems and solutions are expressed in familiar mathematical notation"
+---
+# Introduction to MATLAB
```
Comment (Ajay-Dhangar) [inline docs/MATLAB/Introduction.md:9]: remove this line

File: docs/MATLAB/Syntax.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
+---
+id: MATLAB Syntax
```
Comment (Ajay-Dhangar) [inline docs/MATLAB/Syntax.md:2]: update `id` for good practice

File: docs/MATLAB/Syntax.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+title: Syntax
+sidebar_label: MATLAB Syntax
+sidebar_position: 2
+tags: [MATLAB, Syntax]
+description: In this tutorial, you will learn about Syntax in MATLAB.
+---
+#  MATLAB Syntax
```
Comment (Ajay-Dhangar) [inline docs/MATLAB/Syntax.md:9]: remove this line

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
@Khushi-Pushkar You need to update you `id` value because we are using for `url` or routing so, you can try to write in good format for SEO and best practice and also, removed `#` tag content or may be replace `#` to `##` and use your content **And apply in all files**

**Comment by Khushi-Pushkar** [comment]
could you please advise on how to stop or delete this file (package-lock.json)? I have not made any changes to it, and I would like to take precautions to prevent this issue in the future.

**Comment by Ajay-Dhangar** [comment]
> ould you please advise on how to stop or delete this file (package-lock.json)? I have not made any changes to it, and I would like to take precautions to prevent this issue in the future. If possible, I would greatly appreciate it if you could change the level 1 to level 2 or 3. Thank you very much.

don't worry I will check out and resolve issues

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**